config/flask.py

```diff
@@ -0,0 +1,79 @@
+"""Wrapper around Flask configuration variables."""
+
+from __future__ import annotations
+
+__LICENSE__ = """
+Copyright 2019 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+# XXX: maybe refactor all this to create a flask Config object
+
+import logging
+import random
+import string
+import os
+
+from config.base import config, ConfigurationError
+
+if 'flask' not in config:
+    raise ConfigurationError(
+        'Section [flask] is missing in the configuration.')
+
+USER_SECTION = 'flask'
+
+debug = config.getboolean(USER_SECTION, 'debug', fallback=None)
+if debug is None:
+    logging.error(
+        'Option debug in the section [flask] is unset; '
+        'falling back to prod mode.')
+    debug = False
+
+port = config.getint(USER_SECTION, 'port', fallback=8080)
+
+if debug:
+    hostname = f'127.0.0.1:{port}'
```

Comment:
What if we want the debug mode, with the hostname set?

Reply:
That needs to be revisited anyway, although it's a heated place: we need to be consistent between the discord API and the port here. I'll change that approach on future PRs, because we definitely want to support that moving forward.

Reply:
OK, consider creating an issue about it :)